doc/api/fs.md

```diff
@@ -340,6 +340,81 @@ fs.access('/etc/passwd', fs.constants.R_OK | fs.constants.W_OK, (err) => {
 });
 ```
 
+`fs.access()` should not be used to check if a file exists before using it
+with `fs.open()`, `fs.readFile()` or `fs.writeFile()`. Doing so introduces a
+race condition since other processes may change the file's state between the
+two calls. Instead, user code should open/read/write the file directly and
+handle the error raised if the file is non-existent.
+
+For example:
+
+```js
+// BAD WRITE
+fs.access('myfile', (err) => {
+  if (!err) {
+    console.error('myfile already exists');
+    return;
+  } else {
+    fs.open('myfile', 'wx', (err, fd) => {
+      if (err) throw err;
+      writeMyData(fd);
+    });
+  }
+});
+
+// BETTER WRITE
+fs.open('myfile', 'wx', (err, fd) => {
+  if (err) {
+    if (err.code === "EEXIST") {
+      console.error('myfile already exists');
+      return;
+    } else {
+      throw err;
+    }
+  }
+  writeMyData(fd);
+});
+
+// BAD READ
+fs.access('myfile', (err) => {
+  if (err) {
+    if (err.code === "ENOENT") {
+      console.error('myfile does not exist');
+      return;
+    } else {
+      throw err;
+    }
+  } else {
+    fs.open('myfile', 'r', (err, fd) => {
+      if (err) throw err;
+      readData(fd);
+    });
+  }
+});
+
+// BETTER READ
+fs.open('myfile', 'r', (err, fd) => {
+  if (err) {
+    if (err.code === "ENOENT") {
+      console.error('myfile does not exist');
+      return;
+    } else {
+      throw err;
+    }
+  } else {
+    readMyData(fd);
+  }
+});
+```
+
+The "bad" examples above check for accessibility and then use the file; the
+"better" examples are better because they use the file directly and handle the
+error, if any.
+
+In general, you should only check for the accessibility of a file if you don't
```

Comment:
In general, please avoid the use of `you` in documentation. That said, I'm not certain this last paragraph even needs to be here.